devito/memory.py

```diff
@@ -8,10 +8,9 @@
 import numpy as np
 from sympy import Eq
 
-from devito.dimension import t
 from devito.logger import error
 from devito.tools import convert_dtype_to_ctype
-
+import devito
```

Comment:
This is useful information, perhaps it's useful to file an issue with exactly this description to remember about these circular dependencies